crates/uv-interpreter/src/managed/find.rs

```diff
@@ -1,47 +1,140 @@
+use core::fmt;
+use fs_err as fs;
 use std::collections::BTreeSet;
 use std::ffi::OsStr;
-use std::path::PathBuf;
+use std::io;
+use std::path::{Path, PathBuf};
 use std::str::FromStr;
 
-use once_cell::sync::Lazy;
-use tracing::debug;
-
-use uv_fs::Simplified;
+use uv_state::{StateBucket, StateStore};
 
 use crate::managed::downloads::Error;
 use crate::platform::{Arch, Libc, Os};
 use crate::python_version::PythonVersion;
 
-/// The directory where Python toolchains we install are stored.
-pub static TOOLCHAIN_DIRECTORY: Lazy<Option<PathBuf>> =
-    Lazy::new(|| std::env::var_os("UV_BOOTSTRAP_DIR").map(PathBuf::from));
-
-pub fn toolchains_for_current_platform() -> Result<impl Iterator<Item = Toolchain>, Error> {
-    let platform_key = platform_key_from_env()?;
-    let iter = toolchain_directories()?
-        .into_iter()
-        // Sort "newer" versions of Python first
-        .rev()
-        .filter_map(move |path| {
-            if path
-                .file_name()
-                .map(OsStr::to_string_lossy)
-                .is_some_and(|filename| filename.ends_with(&platform_key))
-            {
-                Toolchain::new(path.clone())
-                    .inspect_err(|err| {
-                        debug!(
-                            "Ignoring invalid toolchain directory {}: {err}",
-                            path.user_display()
-                        );
+/// A collection of installed Python toolchains.
+#[derive(Debug, Clone)]
+pub struct InstalledToolchains {
+    /// The path to the top-level directory of the installed toolchains.
+    root: PathBuf,
+}
+
+impl InstalledToolchains {
+    /// A directory for installed toolchains at `root`.
+    pub fn from_path(root: impl Into<PathBuf>) -> Result<Self, io::Error> {
+        Ok(Self { root: root.into() })
```

Comment:
I've gone back and forth on `Into<PathBuf>`, it's sort of a hidden allocation if you pass in `&Path` I think? Not sure what's best.

Reply:
👍 just going to stick with consistency for now but yeah that makes sense as a concern